client/client.go

```diff
@@ -1,18 +1,41 @@
 package main
 
 import (
-	"fmt"
+	"bufio"
+    "fmt"
+    "os"
 	"log"
+	"math/big"
+	"encoding/json"
 
 	"github.com/gorilla/websocket"
 )
 
 type Message struct {
-	Contents string `json:"contents"`
+	Type string `json:"type"`
```

Comment:
If type should be limited to the only a set of values, using const is more acceptable as it easier to manage in code